### PR TITLE
- Fixes zonemaster/zonemaster-engine#595

### PR DESCRIPTION
**Pull request by vlevigneron**
Add [ZONE10](https://github.com/zonemaster/zonemaster/blob/develop/docs/specifications/tests/Zone-TP/zone10.md) implementation.
Non regression test cases are missing. Not easy to create zones with multiple or wrong SOA. Bind do not load that kind of zone.

Resolves issue #595.

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/Zone.pm [hunk 1]
```diff
@@ -628,6 +654,66 @@ sub zone09 {
...
+                            ns      => $ns->name->string,
+                            address => $ns->address->short,
+                            owner   => $soa[0]->owner,
+                            name    => $name->fqdn,
+                        }
+                      );
+                }
+            }
+            else {
+                push @results, info( NO_RESPONSE_SOA_QUERY => {} );
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Zone.pm]: We have a problem here. The message (message tag) is used for different test cases, but defined differently. I have created an update on Zone10 to replace the message NO_RESPONSE_SOA_QUERY with something unique for Zone10. See zonemaster/zonemaster#802

Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Zone.pm]: Fixed in https://github.com/zonemaster/zonemaster-engine/pull/606/commits/037b87b3f15156c3d3fee7f112e1f17d9f34a109

**Comment by matsduf** [comment]
We need PR zonemaster/zonemaster#802 to be approved and merged before this PR can be approved since it now is based on the change by zonemaster/zonemaster#802.

File: lib/Zonemaster/Engine/Test/Zone.pm [hunk 2]
```diff
@@ -217,6 +227,14 @@ Readonly my %TAG_DESCRIPTIONS => (
...
+        __x    # MULTIPLE_SOA
+          'Nameserver {ns}/{address} responds with multiple ({count}) SOA records on SOA queries.', @_;
+    },
+    NO_RESPONSE => sub {
+        __x    # NO_RESPONSE
+          'Nameserver {ns}/{address} did not respond.', @_;
+    },
     NO_RESPONSE_SOA_QUERY => sub {
         __x    # NO_RESPONSE_SOA_QUERY
           'No response from nameserver(s) on SOA queries.';
```
[the remark this answers is not in the document]
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Zone.pm]: Fixed in https://github.com/zonemaster/zonemaster-engine/pull/606/commits/037b87b3f15156c3d3fee7f112e1f17d9f34a109

**Pull request merged**